Updated readme (#23)
mattermost/mattermost-plugin-user-survey: README.md

> # Mattermost Plugin User Survey

Reviewer: ```suggestion
# Mattermost User Survey Plugin
```

> ### An on-prem plugin for managing surveys withing a Mattermost installation

Reviewer: ```suggestion
### A self-hosted plugin for managing surveys withing a Mattermost installation
```

Reviewer: Is this text intended to be a heading?

> This plugin allows admins to configure, customise, schedule and manage surveys for Mattermost users.

Reviewer: ```suggestion
This plugin allows admins to configure, customize, schedule, and manage surveys for their Mattermost users.
```

> The responses are stored on the Mattermost installation itself, completely on-premise. A CSV report can be generated

Reviewer: ```suggestion
All responses are stored within your Mattermost deployment. You can generate a CSV report
```

for a survey to review the feedback received and analyze trends in NPS scores.

## Features

> * Admins can schedule surveys to start after a specific date and time

Reviewer: ```suggestion
* Admins can schedule surveys to begin on a specific date and a specific time.
```

> * Admins can configure how long does a survey last

Reviewer: ```suggestion
* Admins can configure how long each survey lasts.
```

> * Members of specific teams can be excluded from a survey

Reviewer: ```suggestion
* Members of specific teams can be excluded from a survey.
```

> * Welcome message of the survey can be customised

Reviewer: ```suggestion
* Admins can customize the Welcome message for each survey.
```

> * Currently only one question can be customised with ability to add more questions of different kinds soon to follow.

Reviewer: ```suggestion
* Admins can customize one question. Additional question customization is planned in a future iteration. 
```

> * A report can generated for each survey that includes the NPS score of the survey and all the responses which the users submitted.

Reviewer: ```suggestion
* Admins can generate a report for each survey that includes NPS scores and user responses.
```

<img src="docs/readme/demo_image.png?raw=true" alt="user survey demo screenshot"/>

## Development

### Setup

Make sure you have the following components installed:

- Go - v1.21 - [Getting Started](https://golang.org/doc/install)
  > **Note:** If you have installed Go to a custom location, make sure the `$GOROOT` variable is set properly.
  Refer [Installing to a custom location](https://golang.org/doc/install#install).

- Node.JS - v18.18

- Make

> You also want to have the environment variable `MM_SERVICESETTINGS_ENABLEDEVELOPER="true"` set as otherwise the plugin

Reviewer: ```suggestion
You also want to have the environment variable `MM_SERVICESETTINGS_ENABLEDEVELOPER="true"` set, otherwise the plugin
```

> will be compiled for Linux, Windows and Darwin ARM64 and x64 architecture every single time. Setting

Reviewer: ```suggestion
will be compiled for Linux, Windows, and Darwin ARM64 and x64 architecture every single time. Setting
```

the `MM_SERVICESETTINGS_ENABLEDEVELOPER` to `true` makes the plugin compile and build only for the OS and architecture
you are building on.

> In your mattermost config, make sure that `PluginSettings.EnableUploads` is `true`, and `FileSettings.MaxFileSize` is

Reviewer: ```suggestion
In your Mattermost configuration file, ensure that `PluginSettings.EnableUploads` is set to `true`, and `FileSettings.MaxFileSize` is
```

> large enough to accept the plugin bundle (eg `256000000`)

Reviewer: ```suggestion
set to a large enough value to accept the plugin bundle (eg `256000000`).
```

### Building the plugin

> Run the following command in the plugin repo to prepare a compiled, distributable plugin zip:

Reviewer: ```suggestion
Run the following command in the plugin repository to prepare a compiled, distributable plugin ZIP file:
```
